---
title: "THRA Isoform 1 and Isoform 2 Relative Expression"
output:
  html_document:
    toc: True
    toc_float: True
    df_print: paged
---

INTRODUCTION to the Experiment

	A total of 148 solid tissue samples representing 20 organs were taken from post-mortal human healthy donors killed in road accidents no later than 36 hours after death. Prior to RNA extraction solid normal tissue samples were stored either in RNAlater or as Formalin-Fixed Paraffin-Embedded (FFPE) blocks. Blood and bone marrow samples were taken from 6 and 11 healthy volunteers respectively and subjected immediately to RNA extraction. Following ribosomal RNA depletion and RNA libraries construction transcription profiles of the samples were obtained using Illumina HiSeq 3000.
First-Stranded single-end reads. 50bp. Read counts of THRA exon 9a (Chr17:40089333) and exon 9b (Chr17:40089334)

```{r}
#load libraries
library(tidyverse)
library(readr)
library(ggplot2)
library(plotly)
library(matrixStats)
library(ggrepel)
library(scales)
library(readxl)
library(dplyr)

```


```{r}
## set paths for output figure
path_plots <- "~/OneDrive - Charité - Universitätsmedizin Berlin/Schuelke_Lab/EG15_RNA_Seq/GEO_GSE120795/THRA_isoform_Expression_Coverage"

## load metadata file

metadata <- read.delim("metadata_GEO_GSE120795.txt", sep=",")
metadata_tissues <- select(metadata, Run, "Sample" =source_name)


##Load bedtools outputs
file_links_bedtools <- list.files(path= "~/OneDrive - Charité - Universitätsmedizin Berlin/Schuelke_Lab/EG15_RNA_Seq/GEO_GSE120795/bedtools_outputs" , pattern = "*.txt", full.names=T)


# initialize an empty dataframe
data_bedtools <- data_frame("V1"=character(),
                   "V2"=integer(),
                   "V3"=integer(),
                   "V4"=character(),
                   "V5"=double(),
                   "Run"=character()) 

for (x in file_links_bedtools) {
  table_sample <- read.delim(x, header = FALSE) ## read table
  basename_sample <- basename(x)
  table_sample <- mutate(table_sample, "Run"=basename_sample)
  sum_column <- table_sample$V5
  
  # if statement to filter out samples that have 0 coverage in both exon 9a and 9b
  if (sum(sum_column)  > 0) {
    data_bedtools <- bind_rows(data_bedtools, table_sample)
  }
    
  
}

#extract the SSR number and merge with the Sample name
data_bedtools <- data_bedtools %>%
  separate("Run","Run", "_") %>%
  left_join(metadata_tissues, by="Run")

data_bedtools

##Load mosdepth outputs

file_links_mosdepth <- list.files(path= "~/OneDrive - Charité - Universitätsmedizin Berlin/Schuelke_Lab/EG15_RNA_Seq/GEO_GSE120795/mosdepth_outputs" , pattern = "*.bed", full.names=T)


# initialize an empty dataframe
data_mosdepth <- data_frame("V1"=character(),
                   "V2"=integer(),
                   "V3"=integer(),
                   "V4"=character(),
                   "V5"=double(),
                   "Run"=character()) 

for (x in file_links_mosdepth) {
  table_sample <- read.delim(x, header = FALSE) ## read table
  basename_sample <- basename(x)
  table_sample <- mutate(table_sample, "Run"=basename_sample)
  sum_column <- table_sample$V5
  
  # if statement to filter out samples that have 0 coverage in both exon 9a and 9b
  if (sum(sum_column)  > 0) {
    data_mosdepth <- bind_rows(data_mosdepth, table_sample)
  }
    
  
}

#extract the SSR number and merge with the Sample name
data_mosdepth <- data_mosdepth %>%
  separate("Run","Run", "_")%>%
  left_join(metadata_tissues, by="Run")

data_mosdepth

```

```{r}
## Let's rearrange the data in a useful way
## I will keep track of Run as the replicate identifier
data_mosdepth <- data_mosdepth %>%
  select(Sample, Run, Isoform = V4, Reads_count =V5) %>%
  spread(key="Isoform", value="Reads_count") %>%
  mutate("Package" = as.factor("mosdepth")) 

data_mosdepth

data_bedtools <- data_bedtools %>%
  select(Sample,Run, Isoform = V4, Reads_count =V5) %>%
  spread(key="Isoform", value="Reads_count")%>%
  mutate("Package" = as.factor("bedtools"))

data_bedtools
```

```{r}
## Let's merge the two dataframes into one

dataset <- full_join(data_mosdepth, data_bedtools)

dataset
```



```{r}
## Let's calkculate THRA1 (counts of 9b) and THRA2 (9a-9b)

dataset <- dataset %>%
  mutate("THRA1"= dataset$`9b`) %>%
  mutate("THRA2"=dataset$`9a`-dataset$`9b`) %>%
  rename("Read_counts_9a" = "9a") %>%
  rename("Read_counts_9b" = "9b") %>%
  relocate(Package, .after=Sample)

dataset

```
```{r}
## Let's add the final calculations

dataset_final <- dataset %>%
  mutate("delta_A1vsA2" = THRA1 - THRA2) %>%
  mutate("THRA1_higher" = THRA1 > THRA2)

dataset_final
  
```



```{r}

###
#####
####
###
###
###TEEEEEESSST
### Lets compare the read counts of exon 9a between the 2 tools
graph_data <- dataset_final %>%
  select(Sample, Package, Read_counts_9a)


ggplot(graph_data, aes(x = Sample, y = Read_counts_9a, fill = Package)) +
  geom_boxplot( width=0.5, position=position_dodge(width=0.7)) +
  scale_y_continuous("Read counts") +
  scale_fill_manual("", values = c("bedtools" = "darksalmon", "mosdepth" = "#56B4E9")) +
  ggtitle("Tools comparison - Exon 9a - Max values") +
  theme_light(base_size = 12) +
  scale_x_discrete(guide = guide_axis(angle = -35)) +
  theme(axis.title.x=element_blank())

#ggplotly()

## save plot
ggsave("Read_counts_9a_max_tool_comparison.png" , device=png, dpi = 600, path = path_plots, bg = "transparent", width = 20, height = 15, units = "cm" )
```


```{r}
### Lets compare the read counts of exon 9a between the 2 tools
graph_data <- dataset_final %>%
  select(Sample, Package, Read_counts_9a)


ggplot(graph_data, aes(x = Sample, y = Read_counts_9a, fill = Package)) +
  geom_col( width=0.5, position=position_dodge(width=0.7)) +
  scale_y_continuous("Read counts") +
  scale_fill_manual("", values = c("bedtools" = "darksalmon", "mosdepth" = "#56B4E9")) +
  ggtitle("Tools comparison - Exon 9a - Max values") +
  theme_light(base_size = 12) +
  scale_x_discrete(guide = guide_axis(angle = -35)) +
  theme(axis.title.x=element_blank())

#ggplotly()

## save plot
ggsave("Read_counts_9a_max_tool_comparison.png" , device=png, dpi = 600, path = path_plots, bg = "transparent", width = 20, height = 15, units = "cm" )
```


```{r}
### Lets do the same with mean values
graph_data_mean <- graph_data %>%
  group_by(Sample, Package) %>%
  summarize(Read_counts_9a=mean(Read_counts_9a))


ggplot(graph_data_mean, aes(x = Sample, y = Read_counts_9a, fill = Package)) +
  geom_col( width=0.5, position=position_dodge(width=0.7)) +
  scale_y_continuous("Read counts") +
  scale_fill_manual("", values = c("bedtools" = "darksalmon", "mosdepth" = "#56B4E9")) +
  ggtitle("Tools comparison - Exon 9a - Mean values") +
  theme_light(base_size = 12) +
  scale_x_discrete(guide = guide_axis(angle = -35)) +
  theme(axis.title.x=element_blank())

#ggplotly()

## save plot
ggsave("Read_counts_9a_mean_tool_comparison.png" , device=png, dpi = 600, path = path_plots, bg = "transparent", width = 20, height = 15, units = "cm" )
```



```{r}
### Lets compare the read counts of exon 9b between the 2 tools
graph_data <- dataset_final %>%
  select(Sample, Package, Read_counts_9b)


ggplot(graph_data, aes(x = Sample, y = Read_counts_9b, fill = Package)) +
  geom_col( width=0.5, position=position_dodge(width=0.7)) +
  scale_y_continuous("Read counts") +
  scale_fill_manual("", values = c("bedtools" = "darksalmon", "mosdepth" = "#56B4E9")) +
  ggtitle("Tools comparison - Exon 9b - Max values") +
  theme_light(base_size = 12) +
  scale_x_discrete(guide = guide_axis(angle = -35)) +
  theme(axis.title.x=element_blank())

#ggplotly()
## save plot
ggsave("Read_counts_9b_total_tool_comparison.png" , device=png, dpi = 600, path = path_plots, bg = "transparent", width = 20, height = 15, units = "cm" )
```


```{r}
### Lets do the same with mean values
graph_data_mean <- graph_data %>%
  group_by(Sample, Package) %>%
  summarize(Read_counts_9b=mean(Read_counts_9b))


ggplot(graph_data_mean, aes(x = Sample, y = Read_counts_9b, fill = Package)) +
  geom_col( width=0.5, position=position_dodge(width=0.7)) +
  scale_y_continuous("Read counts") +
  scale_fill_manual("", values = c("bedtools" = "darksalmon", "mosdepth" = "#56B4E9")) +
  ggtitle("Tools comparison - Exon 9b - Mean values") +
  theme_light(base_size = 12) +
  scale_x_discrete(guide = guide_axis(angle = -35)) +
  theme(axis.title.x=element_blank())

#ggplotly()
## save plot
ggsave("Read_counts_9b_mean_tool_comparison.png" , device=png, dpi = 600, path = path_plots, bg = "transparent", width = 20, height = 15, units = "cm" )
```


```{r}
### heatmap with continuos delta THRA1vsA2 using mosdepth
heatmap_data <- filter(dataset_final, Package == "mosdepth")

y_max_mosdepth <- max(heatmap_data$delta_A1vsA2)
y_min_mosdepth <- min(heatmap_data$delta_A1vsA2)

ggplot(heatmap_data, aes(1, reorder(Sample,delta_A1vsA2), fill=delta_A1vsA2)) +
  geom_tile()+
  ggtitle("THRA isoform expression pattern (THRA1/THRA2) - mosdepth") +
  theme_light(base_size = 12)+
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  scale_fill_gradient2('delta_reads', limits=c(y_min_mosdepth, y_max_mosdepth), breaks = c( y_min_mosdepth, 0, y_max_mosdepth),  low = "#56B4E9", high = "darksalmon", guide="colorbar")

ggsave("Heatmap_THRA1vsA2_mosdepth.png" , device=png, dpi = 600, path = path_plots, bg = "transparent", width = 20, height = 15, units = "cm" )
```

```{r}
### heatmap with continuos delta THRA1vsA2 using bedtools
heatmap_data <- filter(dataset_final, Package == "bedtools")

y_max_bedtools <- max(heatmap_data$delta_A1vsA2)
y_min_bedtools <- min(heatmap_data$delta_A1vsA2)

ggplot(heatmap_data, aes(1, reorder(Sample,delta_A1vsA2), fill=delta_A1vsA2)) +
  geom_tile()+
  ggtitle("THRA isoform expression pattern (THRA1/THRA2) - bedtools") +
  theme_light(base_size = 12)+
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  scale_fill_gradient2('delta_reads', limits=c(y_min_bedtools, y_max_bedtools), breaks = c( y_min_bedtools, 0, y_max_bedtools),  low = "#56B4E9", high = "darksalmon", guide="colorbar")

ggsave("Heatmap_THRA1vsA2_bedtools.png" , device=png, dpi = 600, path = path_plots, bg = "transparent", width = 20, height = 15, units = "cm" )
```



```{r}
### heatmap with AUT AUT delta THRA1vsA2 using mosdepth
heatmap_data <- filter(dataset_final, Package == "mosdepth")

ggplot(heatmap_data, aes(1, reorder(Sample,delta_A1vsA2), fill=THRA1_higher)) +
  geom_tile()+
  ggtitle("THRA isoform expression pattern (THRA1/THRA2) - mosdepth") +
  theme_light(base_size = 12)+
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  scale_fill_manual(values = c("TRUE"= "darksalmon", "FALSE" = "#56B4E9"))

ggsave("Heatmap_THRA1vsA2_AUT_AUT_mosdepth.png" , device=png, dpi = 600, path = path_plots, bg = "transparent", width = 20, height = 15, units = "cm" )
```

```{r}
### heatmap with AUT AUT delta THRA1vsA2 using bedtools
heatmap_data <- filter(dataset_final, Package == "bedtools")

ggplot(heatmap_data, aes(1, reorder(Sample,delta_A1vsA2), fill=THRA1_higher)) +
  geom_tile()+
  ggtitle("THRA isoform expression pattern (THRA1/THRA2) - bedtools") +
  theme_light(base_size = 12)+
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  scale_fill_manual(values = c("TRUE"= "darksalmon", "FALSE" = "#56B4E9"))

ggsave("Heatmap_THRA1vsA2_AUT_AUT_bedtools.png" , device=png, dpi = 600, path = path_plots, bg = "transparent", width = 20, height = 15, units = "cm" )
```



```{r}
###Lets compare the methods

ggplot(dataset_final, aes(Package, reorder(Sample,`delta_A1vsA2`) , fill=`delta_A1vsA2`)) +
  geom_tile()+
  ggtitle("Delta comparison between methods (THRA1/THRA2)") +
  theme_light(base_size = 12)+
  theme(axis.title.y=element_blank()) +
  theme(axis.title.x=element_blank()) +
  scale_fill_gradient2('delta', breaks = c(min(y_min_bedtools, y_min_mosdepth), 0,  max(y_max_bedtools,y_max_mosdepth )),  low = "#56B4E9", high = "darksalmon", guide="colorbar")


ggsave("Heatmap_methods_comparison.png" , device=png, dpi = 600, path = path_plots, bg = "transparent", width = 20, height = 15, units = "cm" )
```

